core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -24,6 +24,9 @@
   private TableProperties() {
   }
 
+  private static final String WRITE_PREFIX = "write.";
```

Comment:
Any other ideas are welcome.

Reply:
I haven't got time to read the writer factory changes, will do that during the weekend. 

For the table properties, one thing immediately came to my mind was that I will no longer be able to search for a property based on its string. Maybe it's better to just continue to rewrite `write.` and `write.delete` for each property, and new people editing this file can simply follow the convention. In that way the class is much cleaner and more readable.

Reply:
Well, you are right. I was debating this for a while myself. I was afraid we may miss a typo in the config name after the prefix but I think agree about using separate variables. We just need to make sure reviewers double check them. Having these private prefixes also seem a little bit weird.